src/cpptools.ts

```diff
@@ -485,12 +485,20 @@ export class CppConfigurationProvider implements cpptools.CustomConfigurationPro
         }
 
         if (sysroot) {
-            if (!useFragments) {
+            if (useFragments) {
+                // Pass sysroot (without quote added) as the only compilerArgs for CppTools API V6 and above.
+                flags.push(`--sysroot=${sysroot}`);
```

Comment:
didn't we want to add this to `compileCommandFragments` instead?

Reply:
I'm just switching the order of the branches.  I'm not changing the implementation in this PR.  This works the way it is.